test/concerns/sti_support_test.rb

```diff
@@ -35,4 +35,26 @@ def test_sti_support_with_from_subclass
       Object.send :remove_const, 'SubclassWithAncestry'
     end
   end
+
+  def test_sti_support_for_counter_cache
+    AncestryTestDatabase.with_model :counter_cache => true, :extra_columns => {:type => :string} do |model|
+      Object.const_set 'Subclass3', Class.new(model)
```

Comment:
Is there a reason why you did not use `Subclass1`?

If there were issues with reusing those class names, please let me know and I'll fix any issues.

Reply:
Oh yes sorry I should have mentioned that, using the same class names was causing the error:

```
ActiveRecord::SubclassNotFound: Invalid single-table inheritance type: Subclass1 is not a subclass of Subclass1
```

Reply:
My guess is that Rails is caching the constant name in the previous spec but it's not being wiped after... not sure how that part of Rails works though